Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -422,7 +422,7 @@ private static string GetJobSettingsPath(DirectoryInfoBase jobDirectory)
 
         protected TJobStatus GetStatus<TJobStatus>(string statusFilePath) where TJobStatus : class, IJobStatus, new()
         {
-            return JobLogger.ReadJobStatusFromFile<TJobStatus>(TraceFactory, statusFilePath) ?? new TJobStatus();
+            return JobLogger.ReadJobStatusFromFile<TJobStatus>(TraceFactory, statusFilePath);
```

Comment:
No sure I understand why it's wrong to return a `new` object when `ReadJobStatusFromFile` returns null.